src/BenchmarkDotNet/Code/ArrayParam.cs

```diff
@@ -7,6 +7,12 @@
 
 namespace BenchmarkDotNet.Code
 {
+    internal static class ArrayParam
+    {
+        public static string GetDisplayString(Array array)
+            => $"{array.GetType().GetElementType().Name}[{array.Length}]";
```

Comment:
It will be better to get the type name with the help of `BenchmarkDotNet.Extensions.ReflectionExtensions.GetDisplayName`: this method provides nice presentation for generic classes.

Reply:
Okay, will do.

Reply:
Looks like the class you mentioned is for reflection extensions only. Don't think that moving the method is a good idea. What am thinking about is refactoring smart and array params into a single class.

Reply:
Well, you already call reflection methods (`.GetType().GetElementType().Name`), but your code doesn't work well in the case of generic classes. I'm suggesting to reuse the existing formatting logic for such cases. What's wrong with the usage of `ReflectionExtensions`?

Reply:
`ReflectionExtensions` accept a type and other reflection related classes as `this` parameter, while this one should accept an array. Is it okay to mix such things?

Reply:
@YohDeadfall sorry, I guess, the text of my initial suggestion wasn't clear enough, so let me rephrase. In *this* thread, I just suggested to replace

```cs
            => $"{array.GetType().GetElementType().Name}[{array.Length}]";
```
by
```cs
            => $"{array.GetType().GetElementType().GetDisplayName()}[{array.Length}]";
```
In this case, we will get a better presentation for cases when the element type is generic (e.g., `List<int>[]`).

In *another* thread, I suggested moving the whole `GetDisplayString(Array array)` method to another utility class. It's not clear for me right now which class should we choose, I will think about it later. Once I find a good place, I will refactor it myself.

Reply:
Ah, I see.